File: find_num_even_digit.py
# 1295. Find Numbers with Even Number of Digits

"""
given an array 'nums' of integers, return how many of them contain an even
number of digits.
"""


class Solution(object):
    def findNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return sum(1 for i in nums if (len(str(i)) & 1) == 0)


if __name__ == "__main__":
    obj = Solution()
    print(obj.findNumbers(nums=[12, 345, 2, 6, 7896]))
    print(obj.findNumbers(nums=[555, 901, 482, 1771]))

File: find_num_even_digit.c
// 1295. Find Numbers with Even Number of Digits
#include "leetcode.h"

/*
 * given an array 'nums' of integers, return how many of them contain an even
 * number of digits.
 */

int findNumbers(int *nums, int numsSize) {
  int cnt = 0;
  for (int i = 0; i < numsSize; i++) {
    int curr = 0;
    while (nums[i]) {
      curr++;
      nums[i] /= 10;
    }
    if (!(curr & 1))
      cnt++;
  }
  return cnt;
}

int main() {
  int n1[] = {12, 345, 2, 6, 7896}, n2[] = {555, 901, 482, 1771};
  printf("%d\n", findNumbers(n1, ARRAY_SIZE(n1))); // expect: 2
  printf("%d\n", findNumbers(n2, ARRAY_SIZE(n2))); // expect: 1
}
